crates/ide_db/src/search.rs

```diff
@@ -260,7 +260,7 @@ impl Definition {
             let mut res = source_root.iter().map(|id| (id, None)).collect::<FxHashMap<_, _>>();
 
             let krate = module.krate();
-            for rev_dep in krate.reverse_dependencies(db) {
+            for rev_dep in krate.transitive_reverse_dependencies(db) {
```

Comment:
Is it possible to add a test for this?

Reply:
Will try to make one

Reply:
Seems like a test can't really check this as the first source_root always contains everything already for the tests so it doesnt need to check the `reverse_deps` at all 😕.